### PR TITLE
add config to control read fragment size

### PR DESCRIPTION
**Pull request by superqtqt**
General Changes
* add config to control fragment read buffer size

### CODE REVIEW

**Comment by superqtqt** [comment]
@shixuan-fan can you help me review?

File: presto-main/src/main/java/com/facebook/presto/operator/FileFragmentResultCacheManager.java [hunk 1]
```diff
@@ -281,13 +294,35 @@ public int hashCode()
         }
     }
 
+    private static class CacheValue
```
Comment (shixuan-fan) [inline presto-main/src/main/java/com/facebook/presto/operator/FileFragmentResultCacheManager.java:297]: Maybe we could call this `FileInfo`? Both `path` and `size` is part of file information :)

File: presto-main/src/main/java/com/facebook/presto/operator/FragmentCacheStats.java [hunk 2]
```diff
@@ -35,9 +36,9 @@ public void incrementCacheMiss()
         miss.getAndIncrement();
     }
 
-    public void addInFlightBytes(long bytes)
+    public long addInFlightBytes(long bytes)
```
Comment (shixuan-fan) [inline presto-main/src/main/java/com/facebook/presto/operator/FragmentCacheStats.java:39]: We probably want to change this to `addAndGetInFlightByte`?

File: presto-main/src/main/java/com/facebook/presto/operator/FragmentCacheStats.java [hunk 3]
```diff
@@ -55,6 +56,11 @@ public void decrementCacheEntries()
         cacheEntries.getAndDecrement();
     }
 
+    public long addOutFlightBytes(long bytes)
```
Comment (shixuan-fan) [inline presto-main/src/main/java/com/facebook/presto/operator/FragmentCacheStats.java:59]: ditto

**Review by shixuan-fan — COMMENTED** [review]
Looks good with some minor nits. Let's make the first letter of commit message capital? Also, if we could add some test that would be great :)

**Comment by shixuan-fan** [comment]
Also, you might want to fix the config test

```
Error:  Failures: 
Error:    TestFileFragmentResultCacheConfig.testDefaults:36 Untested attributes: [MaxOutFlightSize]
Error:    TestFileFragmentResultCacheConfig.testExplicitPropertyMappings:69 Untested properties [fragment-result-cache.max-out-flight-size]
```

File: presto-main/src/main/java/com/facebook/presto/operator/FileFragmentResultCacheManager.java (hunk 1 above)
Comment (superqtqt) [reply presto-main/src/main/java/com/facebook/presto/operator/FileFragmentResultCacheManager.java]: fix

File: presto-main/src/main/java/com/facebook/presto/operator/FragmentCacheStats.java (hunk 3 above)
Comment (superqtqt) [reply presto-main/src/main/java/com/facebook/presto/operator/FragmentCacheStats.java]: fix

File: presto-main/src/main/java/com/facebook/presto/operator/FileFragmentResultCacheManager.java [hunk 4]
```diff
@@ -189,20 +193,27 @@ private static void tryDeleteFile(Path path)
     public Optional<Iterator<Page>> get(String serializedPlan, Split split)
     {
         CacheKey key = new CacheKey(serializedPlan, split.getSplitIdentifier());
-        Path path = cache.getIfPresent(key);
-        if (path == null) {
+        FieldInfo value = cache.getIfPresent(key);
```
Comment (shixuan-fan) [inline presto-main/src/main/java/com/facebook/presto/operator/FileFragmentResultCacheManager.java:196]: Change the variable to `fileInfo`.

**Pull request closed without merging**